backend/chromeApiServer/src/test/java/com/canvas/service/helperServices/CanvasClientServiceTest.java

```diff
@@ -1,7 +1,381 @@
+        when(call.execute()).thenReturn(null);
+
+        Assertions.assertThrows(
+                CanvasAPIException.class,
+                () -> canvasClientService.fetchFile(USER_ID, BEARER_TOKEN)
+        );
+    }
+
+    @Test
+    public void testFetchFoldersUnderStudent() throws IOException, CanvasAPIException {
+        when(okHttpClient.newCall(any())).thenReturn(call);
+        when(call.execute()).thenReturn(response);
+
+        JsonNode node = canvasClientService.fetchFoldersUnderStudent(USER_ID, BEARER_TOKEN);
+
+        Assertions.assertEquals(
+                node.get("id").asText(),
+                "fooId"
+        );
+        Assertions.assertEquals(
+                node.get("assignment").get("name").asText(),
+                "fooAssignmentId"
+        );
+        Assertions.assertEquals(
+                node.get("assignment").get("files_url").asText(),
+                "http://fooUrl.foo"
+        );
+    }
+
+    @Test
+    public void testFetchFoldersUnderStudent_throwsCanvasAPIException() throws IOException {
+        when(okHttpClient.newCall(any())).thenReturn(call);
+        when(call.execute()).thenReturn(null);
+
+        Assertions.assertThrows(
+                CanvasAPIException.class,
+                () -> canvasClientService.fetchFoldersUnderStudent(USER_ID, BEARER_TOKEN)
+        );
+    }
+
+    @Test
+    public void testFetchFolders() throws CanvasAPIException, IOException {
+        when(okHttpClient.newCall(any())).thenReturn(call);
+        when(call.execute()).thenReturn(response);
+
+        JsonNode node = canvasClientService.fetchFolders(USER_ID, BEARER_TOKEN);
+
+        Assertions.assertEquals(
+                node.get("id").asText(),
+                "fooId"
+        );
+        Assertions.assertEquals(
+                node.get("assignment").get("name").asText(),
+                "fooAssignmentId"
+        );
+        Assertions.assertEquals(
+                node.get("assignment").get("files_url").asText(),
+                "http://fooUrl.foo"
+        );
+    }
+
+    @Test
+    public void testFetchFolders_throwsCanvasAPIException() throws IOException {
+        when(okHttpClient.newCall(any())).thenReturn(call);
+        when(call.execute()).thenReturn(null);
+
+        Assertions.assertThrows(
+                CanvasAPIException.class,
+                () -> canvasClientService.fetchFolders(USER_ID, BEARER_TOKEN)
+        );
+    }
+
+    @Test
+    public void testFetchFilesUnderFolder() throws CanvasAPIException, IOException {
+        when(okHttpClient.newCall(any())).thenReturn(call);
+        when(call.execute()).thenReturn(response);
+
+        JsonNode node = canvasClientService.fetchFilesUnderFolder("fooFolder", BEARER_TOKEN);
+
+        // TODO: sample for captor use, do after line coverage
+//        verify(okHttpClient).newCall(requestCaptor.capture());
+//        Request request = requestCaptor.getValue();
+
+        Assertions.assertEquals(
```

Comment:
Typically the expected outcome is first and the result of the program is second.

Reply:
Missed that on these asserts. Will change for next revision